src/main/java/com/hrznstudio/titanium/block/tile/inventory/MultiInventoryHandler.java

```diff
@@ -10,33 +10,47 @@
 import com.hrznstudio.titanium.api.IFactory;
 import com.hrznstudio.titanium.api.client.IGuiAddon;
 import com.hrznstudio.titanium.api.client.IGuiAddonProvider;
+import com.hrznstudio.titanium.block.tile.sideness.ICapabilityHolder;
 import com.hrznstudio.titanium.block.tile.sideness.IFacingHandler;
 import com.hrznstudio.titanium.util.FacingUtil;
 import net.minecraft.item.ItemStack;
+import net.minecraftforge.common.util.LazyOptional;
 import net.minecraftforge.items.ItemStackHandler;
 
 import javax.annotation.Nonnull;
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.LinkedHashSet;
-import java.util.List;
+import java.util.*;
 
 
-public class MultiInventoryHandler implements IGuiAddonProvider {
+public class MultiInventoryHandler implements IGuiAddonProvider, ICapabilityHolder<PosInvHandler, MultiInventoryHandler.MultiInvCapabilityHandler> {
 
     private final LinkedHashSet<PosInvHandler> inventoryHandlers;
+    private final HashMap<FacingUtil.Sideness, LazyOptional<MultiInvCapabilityHandler>> lazyOptionals;
 
     public MultiInventoryHandler() {
         this.inventoryHandlers = new LinkedHashSet<>();
+        this.lazyOptionals = new HashMap<>();
+        lazyOptionals.put(null, LazyOptional.empty());
+        for (FacingUtil.Sideness value : FacingUtil.Sideness.values()) {
+            lazyOptionals.put(value, LazyOptional.empty());
+        }
     }
 
-    public void addInventory(PosInvHandler invHandler) {
+    @Override
+    public void add(PosInvHandler invHandler) {
         this.inventoryHandlers.add(invHandler);
+        rebuildCapability(new FacingUtil.Sideness[]{null});
+        rebuildCapability(FacingUtil.Sideness.values());
+    }
+
+    private void rebuildCapability(FacingUtil.Sideness... sides){
+        for (FacingUtil.Sideness side : sides) {
+            lazyOptionals.put(side, getCapabilityForSide(side).map(multiInvCapabilityHandler -> new MultiInvCapabilityHandler(getHandlersForSide(side))));
+        }
     }
 
-    public MultiInvCapabilityHandler getCapabilityForSide(FacingUtil.Sideness sideness) {
+    private List<PosInvHandler> getHandlersForSide(FacingUtil.Sideness sideness){
         if (sideness == null)
-            return new MultiInvCapabilityHandler(new ArrayList<>(inventoryHandlers));
+            return new ArrayList<>(inventoryHandlers);
```

Comment:
So are brackets required or not required for single line ifs. There is no consistency in which to use. Same line or not?